evalml/pipelines/component_graph.py

```diff
@@ -96,8 +97,17 @@ def fit_features(self, X, y):
             X (pd.DataFrame): The input training data of shape [n_samples, n_features]
             y (pd.Series): The target training data of length [n_samples]
         """
-        self._compute_features(self.compute_order[:-1], X, y, fit=True)
-        return self
+        if len(self.compute_order) <= 1:
+            return X
+
+        component_outputs = self._compute_features(self.compute_order[:-1], X, y=y, fit=True)
+        final_component_inputs = []
+        for parent in self.get_parents(self.compute_order[-1]):
+            parent_output = component_outputs.get(parent, component_outputs.get(f'{parent}.x'))
+            if isinstance(parent_output, pd.Series):
+                parent_output = pd.DataFrame(parent_output, columns=[parent])
+            final_component_inputs.append(parent_output)
+        return pd.concat(final_component_inputs, axis=1)
```

Comment:
Is there a unit test where the input dataframe has a custom index? pd.concat introduce NaNs anywhere the indices don't perfectly match up.

Reply:
I've added one! Since a number of components reset the indices and none should be dropping any rows, the test passed without a problem but it's definitely still a good thing to look out for